R/tm_g_patient_profile.R

```diff
@@ -426,89 +426,112 @@ srv_g_patient_profile <- function(id,
   checkmate::assert_class(data, "tdata")
 
   moduleServer(id, function(input, output, session) {
-    iv <- shinyvalidate::InputValidator$new()
-    iv$add_rule("sl_start_date", shinyvalidate::sv_required())
-    iv$add_rule("lb_var_show", shinyvalidate::sv_required())
-    iv$add_rule("ae_var", shinyvalidate::sv_required())
-    iv$enable()
-
     # only show the check box when domain data is available
-    observeEvent(ae_dataname, {
-      if (!is.na(ae_dataname)) {
-        output$select_ae_output <- renderUI({
-          checkboxInput(
-            session$ns("select_ae"),
-            "ADAE",
-            value = !is.na(ae_dataname)
-          )
-        })
-      }
+    checkboxes <- c(ex_dataname, ae_dataname, rs_dataname, lb_dataname, cm_dataname)
```

Comment:
1. `checkboxes[!is.na(checkboxes)]` - this is a carry-over from the module design. Should I make such deep changes in this issue?
2. `show_xx_plot` are not redundant, the app developer can block a plot from display. I don't know  why they would, but they can.
3. `checkboxGroupInput` - thank you :)

Reply:
1. Why not? Alternatively, you can set all `xx_dataname = NA` as default, because right now if anyone skips `xx_dataname` specification the module will throw an error because `c(ex_dataname, ae_dataname, ...)` will fail. But this wouldn't protect anyone from misspecifying `ex_dataname = "dawid"` as it will throw non-specific error message.
 
2. Check Ctrl+f `show_ex_plot` and see where it's used. Users can't block anything with this argument as it's not consumed at all.